### PR TITLE
Fix/backwards compatible

### PR DESCRIPTION
**Pull request by brnleehng**
- Added deprecated pool config validation

https://github.com/Azure/doAzureParallel/issues/62

### CODE REVIEW

File: R/cluster.R [hunk 1]
```diff
@@ -148,29 +146,42 @@ makeCluster <- function(clusterSetting = "cluster_settings.json", fullName = FAL
...
-  pool <- getPool(pool$name)
+  pool <- rAzureBatch::getPool(poolConfig$name)
 
   if (grepl("AuthenticationFailed", response)) {
     stop("Check your credentials and try again.");
   }
 
   if (grepl("PoolExists", response)) {
-    print("The specified pool already exists. Will use existing pool.")
+    cat("The specified pool already exists. Will use existing pool.", fill = TRUE)
```
Comment (paselem) [inline R/cluster.R]: nitpick: "The specified pool 'POOL_NAME' already exists. Pool 'POOL_NAME' will be used."

File: R/cluster.R [hunk 2]
```diff
@@ -148,29 +146,42 @@ makeCluster <- function(clusterSetting = "cluster_settings.json", fullName = FAL
...
+
+    if (pool$targetDedicatedNodes !=  poolConfig$poolSize$dedicatedNodes || pool$targetLowPriorityNodes != poolConfig$poolSize$lowPriorityNodes) {
```
Comment (paselem) [inline R/cluster.R]: Nitpick. Please keep < 100 lines.

File: R/cluster.R [hunk 3]
```diff
@@ -148,29 +146,42 @@ makeCluster <- function(clusterSetting = "cluster_settings.json", fullName = FAL
...
+      warning("There is a mismatched between cluster config nodes and existing pool target nodes")
```
Comment (paselem) [inline R/cluster.R]: Give more info as to what the mismatch is. This doesn't give the user enough info to address the issue.

File: R/cluster.R [hunk 4]
```diff
@@ -108,32 +108,30 @@ generateClusterConfig <- function(fileName, ...){
...
 
-  if (!is.null(pool$rPackages) && !is.null(pool$rPackages$cran) && length(pool$rPackages$cran) > 0) {
-    installCranCommand <- getPoolPackageInstallationCommand("cran", pool$rPackages$cran)
+  if (!is.null(poolConfig$rPackages) && !is.null(poolConfig$rPackages$cran) && length(poolConfig$rPackages$cran) > 0) {
+    installCranCommand <- getPoolPackageInstallationCommand("cran", poolConfig$rPackages$cran)
   }
 
-  if (!is.null(pool$rPackages) && !is.null(pool$rPackages$github) && length(pool$rPackages$github) > 0) {
-    installGithubCommand <- getPoolPackageInstallationCommand("github", pool$rPackages$github)
+  if (!is.null(poolConfig$rPackages) && !is.null(poolConfig$rPackages$github) && length(poolConfig$rPackages$github) > 0) {
```
Comment (paselem) [inline R/cluster.R]: Nitpick: Line too long

File: R/utility.R [hunk 5]
```diff
@@ -249,6 +249,65 @@ validateClusterConfig <- function(clusterFilePath){
   TRUE
 }
 
+#' Validating cluster configuration files below doAzureParallel version 0.3.2
+validateDeprecatedClusterConfig <- function(clusterFilePath){
```
Comment (paselem) [inline R/utility.R]: Going forward we may support multiple of these. I suggest naming it in a version specific way:
validateDepreciatedClusterConfig_0.3.2

Comment (brnleehng) [inline R/utility.R]: Fixed

File: R/cluster.R (hunk 2 above)
Comment (brnleehng) [inline R/cluster.R]: Reformatted with Rstudio (ctrl + shift + a)

File: R/cluster.R (hunk 1 above)
Comment (brnleehng) [inline R/cluster.R]: Updated message

File: R/cluster.R (hunk 4 above)
Comment (brnleehng) [inline R/cluster.R]: Updated

**Pull request merged**